pgjdbc/src/main/java/org/postgresql/jdbc/PgDatabaseMetaData.java

```diff
@@ -2541,7 +2541,25 @@ public boolean isWrapperFor(Class<?> iface) throws SQLException {
 
   public ResultSet getFunctions(String catalog, String schemaPattern, String functionNamePattern)
       throws SQLException {
-    return getProcedures(catalog, schemaPattern, functionNamePattern);
+    String sql;
+    sql = "SELECT NULL AS FUNCTION_CAT, n.nspname AS FUNCTION_SCHEM, p.proname AS FUNCTION_NAME, "
+            + "d.description AS REMARKS, "
+            + DatabaseMetaData.procedureReturnsResult + " AS FUNCTION_TYPE, "
```

Comment:
@zemian , probably by joining `pg_catalog.pg_proc.prorettype` with `pg_catalog.pg_type.oid` and filter that `pg_catalog.pg_type.typname = 'record'` is `functionReturnsTable` else `functionNoTable` would do the trick.

Or maybe something like this makes sense:
```java
sql = "SELECT NULL AS FUNCTION_CAT, n.nspname AS FUNCTION_SCHEM, "
+ "p.proname AS FUNCTION_NAME, d.description AS REMARKS, "
+ "CASE WHEN (format_type(p.prorettype, null) = 'record') THEN " 
+ DatabaseMetaData.functionReturnsTable + " ELSE " 
+ DatabaseMetaData.functionNoTable + " END AS FUNCTION_TYPE, "
+ "p.proname || '_' || p.oid AS SPECIFIC_NAME "
+ "FROM pg_catalog.pg_proc p "
+ "INNER JOIN pg_catalog.pg_namespace n ON (p.pronamespace=n.oid) "
+ "LEFT JOIN pg_catalog.pg_description d ON (p.oid=d.objoid)"
```

I'm not totally sure that `'record'` type is equivalent to a table always, I guess this whould require more research and testing.